var/spack/repos/builtin/packages/fftw/package.py

```diff
@@ -96,6 +96,12 @@ def setup_build_environment(self, env):
                 'CXXFLAGS', self.spec['llvm-openmp'].headers.include_flags)
             env.append_flags(
                 'LDFLAGS', self.spec['llvm-openmp'].libs.ld_flags)
+        # FFTW first configures libtool without MPI, and later uses it with
+        # MPI. libtool then calls wrong linker to create shared libraries
+        # (it calls `$CC` instead of `$MPICC`), and MPI symbols
+        # remain undefined because `-lmpi` is not passed to the linker.
+        if self.spec.satisfies('+mpi'):
+            env.append_flags('LIBS', '-lmpi')
```

Comment:
If this is really needed, then shouldn't we use something like `self.spec['mpi'].ldlibs`? Probably better to patch fftw's configure script ...

Reply:
It should really not be needed. There are mpich installs without `libmpi.so`, like cray-mpich. The fftw configure script does a better job at detecting mpi, this is incorrectly forcing a choice.

Reply:
From the FFTW3 docs:
https://github.com/FFTW/fftw3/blob/fftw-3.3.10/doc/mpi.texi

> Any implementation of the MPI standard, version 1 or later, should
> work with FFTW.  The @code{configure} script will attempt to
> automatically detect how to compile and link code using your MPI
> implementation.  In some cases, especially if you have multiple
> different MPI implementations installed or have an unusual MPI
> software package, you may need to provide this information explicitly.
> 
> Most commonly, one compiles MPI code by invoking a special compiler
> command, typically @code{mpicc} for C code.  The @code{configure}
> script knows the most common names for this command, but you can
> specify the MPI compilation command explicitly by setting the
> @code{MPICC} variable, as in @samp{./configure MPICC=mpicc ...}.
> @fpindex mpicc
> 
> 
> If, instead of a special compiler command, you need to link a certain
> library, you can specify the link command via the @code{MPILIBS}
> variable, as in @samp{./configure MPILIBS=-lmpi ...}.  Note that if
> your MPI library is installed in a non-standard location (one the
> compiler does not know about by default), you may also have to specify
> the location of the library and header files via @code{LDFLAGS} and
> @code{CPPFLAGS} variables, respectively, as in @samp{./configure
> LDFLAGS=-L/path/to/mpi/libs CPPFLAGS=-I/path/to/mpi/include ...}.

Reply:
See https://github.com/spack/spack/pull/29279#issuecomment-1091815103 as well, looks legit to me:
```suggestion
            env.append_flags('LIBS', self.spec['mpi'].libs.ld_flags)
```